ts/commands/ConfigCommands.ts

```diff
@@ -13,29 +13,52 @@ class ConfigCommands {
     (caporal as any)
       .command('config', 'Set local defaults')
       .option('--profile', 'Metricly profile', /.*/, 'default')
+      .option('--username', 'Metricly Username')
+      .option('--password', 'Metricly Password')
+      .option('--url', 'Metricly URL')
       .action((args, options, logger) => {
         const profileConfig = configService.getConfig()[options.profile] || {};
-        inquirer.prompt([{
-          default: profileConfig.username,
-          message: 'Metricly Username',
-          name: 'username',
-          type: 'input'
-        }, {
-          default: profileConfig.password,
-          message: 'Metricly Password',
-          name: 'password',
-          type: 'password'
-        }, {
-          default: profileConfig.endpoint || 'https://app.metricly.com',
-          message: 'Metricly Endpoint',
-          name: 'endpoint',
-          type: 'input'
-        }]).then((answers) => {
+
+        if (!options.username || !options.password) {
+            inquirer.prompt([{
+              default: profileConfig.username,
```

Comment:
If you want to get _really_ clever you can dynamically build the array of what should be prompted. Something like.
```
const prompts = [{
    default: profileConfig.endpoint || 'https://app.metricly.com',
    message: 'Metricly Endpoint',
    name: 'endpoint',
    type: 'input'
}];
if (!options.username) {
  prompts.push(asd;fjalsdfjad);
}
if (!options.password) {

}
inquirer.prompt(prompts)
```

See what I'm getting at?

Reply:
Nice, this is what i really wanted to do! Ok, I have something to towards for another pass!